src/context/auth-context.js

```diff
@@ -0,0 +1,23 @@
+import { createContext, useContext, useReducer } from "react";
+import { authReducer } from "../reducers";
+
+const defaultValue = { user: "", token: "" };
+
+const AuthContext = createContext(defaultValue);
+
+const AuthProvider = ({ children }) => {
+
+    const initialAuthState = { user: "", token: "" };
```

Comment:
No need of writing same initial state two times, you can just pass defaultValue only as your initialState in your reducer.